server/src/validators/signupValidator.js

```diff
@@ -0,0 +1,27 @@
+import { body } from 'express-validator/check';
+import User from '../database/models/user';
+import { customUserValidatorResult } from '../utils/validateUtils';
+
+const signupValidator = [
+  body('email')
+    .isEmail()
+    .withMessage('must be a valid email')
+    .custom(async (email) => {
+      const user = await User.findUserByField('email', email);
```

Comment:
did you need to create `findUserByField` when `findOne` already exist?

Reply:
Oh right. I am dumb.